uint8 kin_mode
---
bool kin_bool_mode